### ISSUE TITLE
Circular remodel chain

### ISSUE THREAD

**Issue by Javran** [poooi/plugin-navy-album#8]
The problem is not exactly solved as Souya (699) should be the chain base.

There are some speculation on how `api_sort_id` are designated to ships: https://nga.178.com/read.php?tid=14868912, which might walk us towards a solution: for a circular remodel chain, tiebreak by last digit of `api_sort_id`. On an off chance that further tiebreaks are needed, compare `api_sortno` then `api_id` in that order.

Or alternatively tiebreak by `api_sortno` rather than last digit of `api_sort_id`, which is a bit simpler (although for the purpose of just finding the chain base it doesn't matter than much):
![2021-11-02_11-21](https://user-images.githubusercontent.com/1096354/139922969-75a22747-4c9b-4bf7-83a5-3dad965b5e22.png)

Need to evaluation against all known remodel chains and see which one we should go for.

**Comment by Javran** [comment]
Some analysis based on latest api_start2:
It seems both sorting by `api_sortno` and by last digit of `api_sort_id` is good enough for finding the base, however both are not reliable on its own to form roughly accurate remodel chains (which is not an issue since the algorithm is mostly based on `api_aftershipid`), some example of inconsistencies if we were to sort by `api_sortno` (numbers in parentheses are id, last digit of sort_id, sortno):

```
Inconsistent: 山風(457,1,257) 山風改二(588,6,388) 山風改二丁(667,7,467) 山風改(369,2,1449)
> sort_id: 山風(457,1,257) 山風改(369,2,1449) 山風改二(588,6,388) 山風改二丁(667,7,467)
Inconsistent: 矢矧(139,1,139) 矢矧改二(663,6,463) 矢矧改二乙(668,7,468) 矢矧改(307,2,1403)
> sort_id: 矢矧(139,1,139) 矢矧改(307,2,1403) 矢矧改二(663,6,463) 矢矧改二乙(668,7,468)
Inconsistent: 親潮(456,1,256) 親潮改二(670,6,470) 親潮改(362,2,1442)
> sort_id: 親潮(456,1,256) 親潮改(362,2,1442) 親潮改二(670,6,470)
Inconsistent: 加賀(84,1,7) 加賀改二(698,6,405) 加賀改二戊(610,7,410) 加賀改二護(646,8,446) 加賀改(278,2,1378)
> sort_id: 加賀(84,1,7) 加賀改(278,2,1378) 加賀改二(698,6,405) 加賀改二戊(610,7,410) 加賀改二護(646,8,446)
Inconsistent: Gambier Bay(544,1,344) Gambier Bay Mk.II(707,6,458) Gambier Bay改(396,2,1476)
> sort_id: Gambier Bay(544,1,344) Gambier Bay改(396,2,1476) Gambier Bay Mk.II(707,6,458)
Inconsistent: 大鯨(184,1,184) 龍鳳(185,1,185) 龍鳳改(318,2,190) 龍鳳改二戊(883,7,483) 龍鳳改二(888,6,488)
> sort_id: 大鯨(184,1,184) 龍鳳(185,1,185) 龍鳳改(318,2,190) 龍鳳改二(888,6,488) 龍鳳改二戊(883,7,483)
```

In this sense, last digit of `api_sort_id` is "more accurate", but we'll need to tie-break by `api_sortno` for 🐳 (also throwing in `api_id` for good measure).

In addition, it might make more sense that we don't just build remodel-chain, but remodel-DAG (since this **is** a possibility given how current api is structured) - cycles are broken by performing DFS from original ship and skip those that have been visited.

Those are definitely overkill for just finding the original ships, but I want to settle this issue once for all.

**Comment by Javran** [comment]
TL;DR: whenever ordering is required, do last digit of `api_sort_no`, `api_sortno`, then `api_id` in that order, all ascending.
